dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/BaseIdentifiableObject.java

```diff
@@ -725,14 +649,58 @@ else if ( idScheme.is( IdentifiableProperty.ATTRIBUTE ) )
         return null;
     }
 
-    /**
-     * Set legacy sharing collections to null so that the ImportService will
-     * import current object with new Sharing format.
-     */
-    public void clearLegacySharingCollections()
+    // -------------------------------------------------------------------------
+    // Sharing helpers
+    // -------------------------------------------------------------------------
+
+    public void setExternalAccess( boolean externalAccess )
```

Comment:
Those helper methods will be moved to a separate `ShareableIdentifiableObject` class in another PR.